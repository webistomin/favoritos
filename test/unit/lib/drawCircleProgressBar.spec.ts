import Favoritos from '../../../src/favoritos';

describe('Favoritos: drawCircleProgressBar method', () => {
  beforeEach(() => {
    document.head.innerHTML = `<link rel="shortcut icon" href="#">`;
  });

  it('Draw circle progress if "progress" param is correct', () => {
    const lib = new Favoritos();
    expect(() => lib['drawCircleProgressBar'](50)).not.toThrow(TypeError);
    expect(() => lib['drawCircleProgressBar'](50)).not.toThrow(DOMException);
    // @ts-ignore
    expect(lib['iconCanvasContext'].__getEvents()).toMatchSnapshot();
  });
});
