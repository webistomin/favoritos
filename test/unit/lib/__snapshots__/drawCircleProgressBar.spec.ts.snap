// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favoritos: drawCircleProgressBar method Draw circle progress if "progress" param is correct 1`] = `
Array [
  Object {
    "props": Object {
      "value": "18px Helvetica, Arial, sans-serif",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  Object {
    "props": Object {
      "value": "center",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textAlign",
  },
  Object {
    "props": Object {
      "value": "middle",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textBaseline",
  },
  Object {
    "props": Object {
      "x": 1,
      "y": 1,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "scale",
  },
  Object {
    "props": Object {
      "anticlockwise": false,
      "endAngle": 7.853981633974483,
      "radius": 14,
      "startAngle": 4.71238898038469,
      "x": 16,
      "y": 16,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "arc",
  },
  Object {
    "props": Object {
      "anticlockwise": false,
      "endAngle": 7.853981633974483,
      "radius": 14,
      "startAngle": 4.71238898038469,
      "x": 16,
      "y": 16,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "arc",
  },
]
`;
